### ISSUE TITLE
cookbook: shuffling batches

### ISSUE THREAD

**Issue by khegazy** [google/jax-md#51]
When creating batches 

def batch(key):
  steps_per_epoch = no_training_samples // batch_size 
  train_epochs = train_steps // steps_per_epoch
  for s in range(train_epochs):
    key, split = random.split(key)
    permutation = random.shuffle(split, training_samples)
    positions = train_positions[permutation]
    features = train_features[permutation]
    for i in range(0, no_training_samples, batch_size):
      batch_data = (positions[permutation[i:i + batch_size]], 
                    features[permutation[i:i + batch_size]])
      yield batch_data

  
you first permute the training data
      positions = train_positions[permutation]
and then access the permuted data via the permutations
    positions[permutation[i:i + batch_size]],

This effectively shuffles and then unshuffles the data so you never have randomly shuffled batches.
